File: platforms/VS/CSharp_Tests/Exceptions_Tests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using carbon14.FuryUtils;
using System;
using System.Linq;

namespace CSharp_Tests
{
    [TestClass]
    public class Exceptions_Tests
    {
        [TestMethod]
        public void When_GetExceptionCode_is_called_Then_the_correct_code_is_returned()
        {
            Assert.AreEqual(0, FuryException.Code());
        }

        [TestMethod]
        public void When_Get_ExceptionString_is_called_Then_the_correct_string_is_returned()
        {
            Assert.AreEqual("", FuryException.ErrorString());
        }
    }
}

File: platforms/VS/C_Tests/ExceptionsTests.cpp
#include "pch.h"
#include "CppUnitTest.h"
#include "utils.hpp"

#include "../../../include/FuryUtils.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace Exceptions_Tests
{
	TEST_CLASS(Exceptions_Tests)
	{
	public:
		TEST_METHOD(When_GetExceptionCode_is_called_Then_the_correct_value_is_returned) {
			Assert::AreEqual(0, GetExceptionCode(), L"Incorrect error code");
 		}

		TEST_METHOD(When_GetExceptionString_is_called_Then_the_correct_value_is_returned) {
			Assert::AreEqual("", GetExceptionString(), L"Incorrect error message");
		}
	};
}
